src/classes/PS_IntegrationService.cls

```diff
@@ -54,33 +54,143 @@ global class PS_IntegrationService {
+        updateConfigurationRecords(configurationsToUpdateByServiceKey.values());
+    }
+
+    /*******************************************************************************************************
+    * @description Creates a map of Payment_Services_Configuration__c records by their Service_Key__c from
+    * the parameters provided by the caller that need to be upserted.
+    *
+    * @param configParams contains settings, such as the baseURL for connecting to payments services
+    * @return A map of Payment_Services_Configuration__c records by their Service_Key__c
+    */
+    private static Map<String, Payment_Services_Configuration__c> createConfigurationsToUpsert(
+        Map<String, List<Map<String, String>>> configParams) {
 
         if (configParams == null || configParams.isEmpty()) {
             throw new IntegrationServiceException(INVALID_CONFIG_PARAMS);
         }
 
-        List<Payment_Services_Configuration__c> configurations = new List<Payment_Services_Configuration__c>();
+        Map<String, Payment_Services_Configuration__c> configurationsToUpsert =
+            new Map<String, Payment_Services_Configuration__c>();
 
         for (String service : configParams.keySet()) {
             for (Map<String, String> valueMap : configParams.get(service)) {
-                configurations.add(createConfiguration(service, valueMap));
+                Payment_Services_Configuration__c configuration = createConfiguration(service, valueMap);
+                configurationsToUpsert.put(configuration.Service_Key__c, configuration);
             }
         }
 
+        return configurationsToUpsert;
+    }
+
+    /*******************************************************************************************************
+    * @description Queries for existing Payment_Services_Configuration__c records
+    *
+    * @param serviceKeys A set of Payment_Services_Configuration__c.Service_Key__c values
+    */
+    private static Payment_Services_Configuration__c[] getExistingRecords(Set<String> serviceKeys) {
+        return [
+            SELECT Id,
+                Service_Key__c,
+                Key__c,
+                Value__c
+            FROM Payment_Services_Configuration__c
+            WHERE Service_Key__c = :serviceKeys
+        ];
+    }
+
+    /*******************************************************************************************************
+    * @description Applies the existing configuration record ids to the corresponding newly generated
+    * configuration in preparation for an update.
+    *
+    * @param existingConfigurations List of existing Payment_Services_Configuration__c records
+    * @param configurationsToUpsertByServiceKey Map of newly generated Payment_Services_Configuration__c records
+    * from the parameters provided by the caller of 'setConfiguration'.
+    */
+    private static void applyExistingIdsToNewConfigurations(
+        Payment_Services_Configuration__c[] existingConfigurations,
```

Comment:
Even though Arrays and Lists are almost the same in apex, I would consider using List<> instead of [] for consistency. We are using `List<Payment_Services_Configuration__c>` in code above.
This comment applies to some of the methods below.